File: dissertation/examples/jade/src/main/java/honours/Partner.java
package honours;

import jade.core.Agent;
import jade.core.behaviours.CyclicBehaviour;
import jade.lang.acl.ACLMessage;
import jade.util.Logger;

import java.util.logging.Level;

public class Partner extends Agent {
    private final Logger logger = Logger.getJADELogger(this.getClass().getName());

    @Override
    protected void setup() {
        super.setup();
        logger.setLevel(Level.INFO);

        logger.info("[" + getLocalName() + "] Starting");

        addBehaviour(new CyclicBehaviour(this) {
            @Override
            public void action() {
                ACLMessage msg = myAgent.receive();

                if(msg != null) {
                    switch(msg.getContent()) {
                        case "Hello":
                            logger.info("[" + getLocalName() + "] Say hello!");
                            ACLMessage reply = msg.createReply();
                            reply.setContent("Hello");
                            send(reply);
                            break;
                        case "Die":
                            logger.info("[" + getLocalName() + "] is dying");
                            myAgent.doDelete();
                            break;
                        default:
                            break;
                    }

                } else {
                    block();
                }
            }
        });
    }
}

File: dissertation/examples/spade/partner.py
from spade.agent import Agent
from spade.behaviour import CyclicBehaviour

class Partner(Agent):
    class Behaviour(CyclicBehaviour):
        async def run(self):
            msg = await self.receive(timeout=1)

            if msg is not None:
                if msg.body == 'Hello':
                    print("[{}] Say hello!".format(self.agent.jid))
                    reply = msg.make_reply()
                    reply.body = 'Hello'
                    await self.send(reply)
                elif msg.body == 'Die':
                    print("[{}] is dying".format(self.agent.jid))
                    self.kill(exit_code=0)

    async def setup(self):
        print("[{}] Starting".format(self.jid))
        self.behaviour = self.Behaviour()
        self.add_behaviour(self.behaviour)
